### PR TITLE
feat(doc): handle basic reexports

### PR DESCRIPTION
**Pull request by bartlomieju**

### CODE REVIEW

File: cli/doc/parser.rs [hunk 1]
```diff
@@ -48,7 +57,23 @@ impl From<BufferedError> for Vec<Diagnostic> {
   }
 }
 
+pub trait DocFileLoader {
+  fn resolve(
+    &self,
+    specifier: &str,
+    referrer: &str,
```
Comment (bartlomieju) [inline cli/doc/parser.rs:64]: Use stronger types - this is very much alike to `ModuleLoader` - but we don't care about compiled version, just TS

File: cli/doc/parser.rs [hunk 2]
```diff
@@ -48,7 +57,23 @@ impl From<BufferedError> for Vec<Diagnostic> {
...
+  ) -> Result<ModuleSpecifier, OpError> {
+    ModuleSpecifier::resolve_import(specifier, referrer).map_err(OpError::from)
+  }
+
+  fn load_source_code(
```
Comment (bartlomieju) [inline cli/doc/parser.rs:69]: output `SourceFile` and change SWC lexer settings based on media type (short circuit for binary files like `.wasm`)

File: cli/doc/parser.rs [hunk 3]
```diff
@@ -26,9 +26,18 @@ use regex::Regex;
...
 use super::DocNodeKind;
 use super::Location;
+use crate::op_error::OpError;
+use deno_core::ModuleSpecifier;
+
+use futures::Future;
+use std::collections::HashMap;
+use std::pin::Pin;
 
 pub type SwcDiagnostics = Vec<Diagnostic>;
```
Comment (bartlomieju) [inline cli/doc/parser.rs:42]: This type alias is horrible, remove it

File: cli/doc/node.rs [hunk 4]
```diff
@@ -46,6 +46,35 @@ impl Into<Location> for swc_common::Loc {
...
+  // export * from "./path/to/module.js";
+  All,
+  // export * as someNamespace from "./path/to/module.js";
+  Namespace(String),
+  // export default from "./path/to/module.js";
+  Default,
+  // (identifier, optional alias)
+  // export { foo } from "./path/to/module.js";
+  // export { foo as bar } from "./path/to/module.js";
+  Named(String, Option<String>),
```
Comment (ry) [inline cli/doc/node.rs:61]: Use triple slash comments here.

File: cli/doc/parser.rs [hunk 5]
```diff
@@ -22,43 +23,96 @@ use crate::swc_ecma_parser::Session;
...
+use super::node::ModuleDoc;
 use super::DocNode;
 use super::DocNodeKind;
 use super::Location;
 
-pub type SwcDiagnostics = Vec<Diagnostic>;
+#[derive(Clone, Debug)]
+pub struct SwcDiagnosticBuffer {
+  pub diagnostics: Vec<Diagnostic>,
+}
```
Comment (ry) [inline cli/doc/parser.rs:48]: I don't really see how SwcDiagnosticBuffer is related to this patch.. but whatever.

File: cli/doc/tests.rs [hunk 6]
```diff
@@ -732,3 +771,94 @@ fn optional_return_type() {
...
+    }
+  ]);
+  let actual = serde_json::to_value(entries.clone()).unwrap();
+  assert_eq!(actual, expected_json);
+
+  assert!(
+    colors::strip_ansi_codes(super::printer::format(entries).as_str())
+      .contains("function fooFn(a: number)")
+  );
+}
```
Comment (ry) [inline cli/doc/tests.rs:864]: Nice test

File: cli/doc/parser.rs (hunk 5 above)
Comment (bartlomieju) [reply cli/doc/parser.rs:48]: Me neither, this is just some infrastructure needed for `swc`, just renamed to something more descriptive

**Pull request merged**